MVVM Module/Source/Feed Presentation/FeedViewModel.swift

```
@@ -17,16 +17,22 @@ final class FeedViewModel {
 	var title: String {
         Localized.Feed.title
 	}
+    
+    var errorMessage = Localized.Feed.loadError
 
 	var onLoadingStateChange: Observer<Bool>?
 	var onFeedLoad: Observer<[FeedImage]>?
+    var onFeedFailed: Observer<String>?
 	
 	func loadFeed() {
 		onLoadingStateChange?(true)
+        let errorMessage = self.errorMessage
```

Following my previous comment, before starting to load the feed, you can send an error state change event to hide the error:

```
onErrorStateChange?(.none)
```

So the UI state transitions are all centralized in the ViewModel. The UI simply follows the event changes.